corgidrp/walker.py

```diff
@@ -235,16 +237,49 @@ def guess_template(dataset):
         elif image.pri_hdr['VISTYPE'] == "FFIELD":
             recipe_filename = "l1_flat_and_bp.json"
         elif image.pri_hdr['VISTYPE'] == "DARK":
-            recipe_filename = "l1_to_l2a_noisemap.json"
+            _, unique_vals = dataset.split_dataset(exthdr_keywords=['EXPTIME', 'CMDGAIN', 'KGAIN'])
+            if 'ISPC' in image.ext_hdr: # to be added eventually, so check if it's present first
+                if image.ext_hdr['ISPC']:
+                    recipe_filename = "l1_to_l2b_pc_dark.json"
+                elif len(unique_vals) > 1: # darks for noisemap creation
+                    recipe_filename = "l1_to_l2a_noisemap.json"
+                elif len(unique_vals) == 1: # traditional darks
```

Comment:
We shouldn't end on an `if`. We should turn this into an `else` or have an `else` below that raises an Exception. 

Same for the 3 other places you have this structure.